pkg/broker/ingress/multi_topic_decouple_sink.go

```diff
@@ -80,6 +86,13 @@ func (m *multiTopicDecoupleSink) Send(ctx context.Context, broker types.Namespac
 		return err
 	}
 
+	// Check to see if there are any triggers interested in this event. If not, no need to send this
+	// to the decouple topic.
+	// TODO(#1804): remove first check when enabling the feature by default.
+	if m.enableEventFiltering && !m.hasTrigger(ctx, &event) {
```

Comment:
I don't think this needs a metric. 

It might be helpful for us to be able to say 'look how much we saved in money and latency (by sending the upstream ack faster)', but I don't see much utility for users, other than when investigating different Broker implementations.

If users are concerned that events are being dropped when they shouldn't be, then I think our answer is that you need to compare Broker `event_count` and Trigger `event_count`. Because the general case of the problem is 'I am sending events to the Broker, but my Trigger is not receiving them'. And that is a problem regardless of whether other Triggers happen to include that event or not. So distinguishing between it being dropped here or at Fanout shouldn't matter.

If you think it is likely this area will have a bug, then a metric may make it easier to track down. But I would argue adding a debug level log statement would likely be better, as we could include the attributes of the event being dropped and investigate if we think it was a mistake or not.

Reply:
Thanks for weighing in. I added a log message to help debugging just in case it's needed.